fearless/Modules/Staking/SelectValidatorsFlow/SelectedValidatorList/SelectedValidatorListViewController.swift

```diff
@@ -0,0 +1,240 @@
+                .commonEdit(preferredLanguages: selectedLocale.rLanguages)
+        }
+    }
+
+    private func updateProceedButton() {
+        let cellViewModels = viewModel?.cellViewModels ?? []
+
+        let buttonTitle: String
+        let enabled: Bool
+        let fillColor: UIColor
+
+        if cellViewModels.count > selectedValidatorsLimit {
+            enabled = false
+            fillColor = R.color.colorDarkGray()!
+            buttonTitle = R.string.localizable
+                .stakingCustomProceedButtonDisabledTitle(
+                    selectedValidatorsLimit,
+                    preferredLanguages: selectedLocale.rLanguages
+                )
+
+        } else {
+            enabled = true
+            fillColor = R.color.colorAccent()!
+            buttonTitle = R.string.localizable
+                .commonContinue(
+                    preferredLanguages: selectedLocale.rLanguages
+                )
+        }
+
+        rootView.proceedButton.triangularedView?.fillColor = fillColor
+        rootView.proceedButton.imageWithTitleView?.title = buttonTitle
+        rootView.proceedButton.isEnabled = enabled
+    }
+
+    private func updateHeaderView() {
+        guard let viewModel = viewModel else { return }
+        headerView?.bind(
+            viewModel: viewModel.headerViewModel,
+            shouldAlert: viewModel.limitIsExceeded
+        )
+    }
+
+    private func presentValidatorInfo(at index: Int) {
+        presenter.didSelectValidator(at: index)
+    }
+
+    // MARK: - Actions
+
+    @objc private func tapEditButton() {
+        rootView.tableView.setEditing(
+            !rootView.tableView.isEditing,
+            animated: true
+        )
+        updateEditButton()
+        updateProceedButton()
+    }
+
+    @objc private func tapProceedButton() {
+        presenter.proceed()
+    }
+}
+
+// MARK: - Localizable
+
+extension SelectedValidatorListViewController: Localizable {
+    func applyLocalization() {
+        if isViewLoaded {
+            title = R.string.localizable
+                .stakingSelectedValidatorsTitle(preferredLanguages: selectedLocale.rLanguages)
+
+            updateEditButton()
+            updateProceedButton()
+            updateHeaderView()
+        }
+    }
+}
+
+// MARK: - SelectedValidatorListViewProtocol
+
+extension SelectedValidatorListViewController: SelectedValidatorListViewProtocol {
+    func updateViewModel(_ viewModel: SelectedValidatorListViewModel) {
+        self.viewModel = viewModel
+        updateHeaderView()
+        updateProceedButton()
+    }
+
+    func reload(_ viewModel: SelectedValidatorListViewModel) {
+        self.viewModel = viewModel
+        updateProceedButton()
+        rootView.tableView.reloadData()
+    }
+
+    func didRemoveItem(at index: Int) {
+        let indexPath = IndexPath(row: index, section: 0)
+        rootView.tableView.deleteRows(at: [indexPath], with: .left)
+
+        let cellViewModels = viewModel?.cellViewModels ?? []
```

Comment:
So we are making an assumption that updateViewModel is called before this method. I would recommend to rename the method to didChangeViewModel(..., byRemovingItemAt index:).